File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/Testscenario.java
package no.nav.pensjon.vtp.testmodell.repo;

import java.util.Map;
import java.util.Set;

import no.nav.pensjon.vtp.testmodell.inntektytelse.InntektYtelseModell;
import no.nav.pensjon.vtp.testmodell.organisasjon.OrganisasjonModeller;
import no.nav.pensjon.vtp.testmodell.personopplysning.PersonArbeidsgiver;
import no.nav.pensjon.vtp.testmodell.personopplysning.Personopplysninger;
import no.nav.pensjon.vtp.testmodell.util.VariabelContainer;

public interface Testscenario {

    String getTemplateNavn();

    String getId();

    Personopplysninger getPersonopplysninger();

    OrganisasjonModeller getOrganisasjonModeller();

    VariabelContainer getVariabelContainer();

    InntektYtelseModell getSøkerInntektYtelse();

    InntektYtelseModell getAnnenpartInntektYtelse();

    Set<PersonArbeidsgiver> getPersonligArbeidsgivere();
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/Testscenario.kt
package no.nav.pensjon.vtp.testmodell.repo

import no.nav.pensjon.vtp.testmodell.inntektytelse.InntektYtelseModell
import no.nav.pensjon.vtp.testmodell.organisasjon.OrganisasjonModell
import no.nav.pensjon.vtp.testmodell.personopplysning.Personopplysninger
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Testscenario(
        @Id
        val id: String,
        val templateNavn: String,
        val personopplysninger: Personopplysninger,
        val søkerInntektYtelse: InntektYtelseModell?,
        val annenpartInntektYtelse: InntektYtelseModell?,
        val organisasjonModeller: List<OrganisasjonModell>,
        val vars: Map<String, String?>
)
